src/views/AddItem.jsx

```diff
@@ -22,7 +22,8 @@ export function AddItem({ listToken }) {
 		setDaysUntilNextPurchase(7);
 	};
 
-	const handleNameInput = (e) => setItemName(e.target.value);
+	const handleNameInput = (e) =>
+		setItemName(e.target.value.split(/ +/).join(' '));
```

Comment:
Oo cool addition for input validation! Another thing you could consider is `.trim()` (only once it's submitted) to help avoid leading/trailing white space.

Reply:
Hi @itsoliviasparks! Sorry for the delay! Ah yes - I see what you mean! This was an interesting problem to dig into. So in `AddItem.js`, we are passing `itemName` to `addItem` like this: `addItem(listId, { itemName, daysUntilNextPurchase })`. That's really like passing an object like `{itemName: "apple", daysUntilNextPurchase: 7}`.

Then, in `firebase.js`, we have this again `addItem(listId, { itemName, daysUntilNextPurchase })`, but in this case, `{ itemName, daysUntilNextPurchase}` is _DESTRUCTURING_, so basically it's counting on there being a key called `itemName`. So if you tried to pass in the following, it would return undefined, because we're no longer passing a key called `itemName`:
```
const trimmedItemName = itemName.trim()
await addItem(listToken, { trimmedItemName, daysUntilNextPurchase });
```

Instead, you can do this in `AddItem.js`:
```
const addItemToList = async () => {
	const trimmedItemName = itemName.trim()
	try {
		await addItem(listToken, { itemName: trimmedItemName, daysUntilNextPurchase });
...
```

Let me know if you have any questions about it!

Reply:
🤯 Thank you so much!